sapi/fpm/fpm/fpm_php.c

```diff
@@ -180,6 +180,13 @@ char *fpm_php_query_string(void) /* {{{ */
 }
 /* }}} */
 
+char *fpm_php_client(void) /* {{{ */
+{
+	fcgi_request *request = (fcgi_request*) SG(server_context);
+        return FCGI_GETENV(request, "REMOTE_ADDR");
```

Comment:
This relies on the fact that client (e.g. nginx, apache or whatever else) will supply `REMOTE_ADDR` env variable which is usually the case but it's not required. There is only one env require to execute a script in FPM which is `SCRIPT_FILENAME`. I guess the function that you are looking for is `fcgi_get_last_client_ip`.

Reply:
Maybe I'm wrong, but it seems to me that fcgi_get_last_client_ip returns the other side of the FCGI tcp socket (ie. IP address of the webserver which is connecting to PHP-FPM), not the IP of the client that sent HTTP request for that page, which is in REMOTE_ADDR env variable. That is what I would like to see in the status of PHP-FPM page...

Reply:
Well the only client from the FPM point of view is the other end of the socket. I guess you are not actually looking for that information though so my mistake to suggest it.

However I'm not sure if selecting specific env in the status is a good solution. For example why not add also REMOTE_PORT in that case and how we can know what other envs will be useful for someone. Maybe displaying all env variables would make most sense. It just seems a little bit a hack to just display a single specific env.

Reply:
Yes, you are right, it's a hack to make PHP-FPM status more admin-friendly.
Use case is like this:
 - some PHP script has gone wild, using too much resources, executing too long, stuck...
 - admin (or script) wants to drop that connection and ban that client (for a period of time or something)
 - PHP-FPM status has information of PID and Request URI
 - various ps, netstat, /proc, and webserver status magic is needed to find address of the client
 - enhanced PHP-FPM status comes to resque

Besides REMOTE_ADDR it would be useful to see also REMOTE_PORT and HTTP_HOST (like in Apache HTTPD server-status) but not required for this specific use case.

As for the other variables, maybe we could have some configurable variable for list of HTTP headers from environment one would want to be shown in the FPM status? Something like this in php-fpm.conf:
`pm.status_env = HTTP_HOST,REMOTE_PORT,HTTP_X_FORWARDED_FOR,HTTP_X_FORWARDED_PROTO`

I'm happy to try to implement it, if it seems like a good idea...

Reply:
I think it would be useful to add another param like `full` (for example `env`) that would display all supplied env vars for the last FCGI request. That would be a bit more flexible and there wouldn't be a need for an extra config.

Reply:
Yes, that seems cool to me too. I'll try to do it that way, just please tell me if I should do it by pushing more commits to the same branch where I made this pull request, or should I make new branch and make new pull request when it's ready? I wouldn't want to make a mess :)

Reply:
@mnspodrska Sorry for the late reply... :) You can push new commits to this PR and once you are happy with that, just ping me and I will take a look. If there are no issues or objections from others, I will ask you to squash it, push force it in here and then I will merge it.  ;)

Reply:
Sorry completely missed that this got updated and then forgot about this.

Just minor thing that the CS looks wrong here - should use tabs.